### PR TITLE
feat: add granular steps component

### PR DESCRIPTION
**Pull request by daviditsygin**
Added a variant for the `<Steps/>` to display a progress bar for the current step.

`stepProgress` = 28
image
`stepProgress` = 75
image

### CODE REVIEW

**Comment by daviditsygin** [comment]
[react-gears: add granular progress bar to Steps component](https://trello.com/c/BmhJImao/12-react-gears-add-granular-progress-bar-to-steps-component)

File: src/components/Steps/Steps.tsx [hunk 1]
```diff
@@ -33,13 +37,35 @@ const Steps = ({
...
+                  progressPercent + stepProgress
+                }%, ${barColor} ${progressPercent + stepProgress}%)`
+              : renderPrevStepProgress
+              ? `linear-gradient(to right, ${progressColor} ${
+                  stepProgress - progressPercent
+                }%, ${barColor} ${stepProgress - progressPercent}%, ${barColor} 50%, ${
+                  index === steps.length - 1 ? 'transparent' : barColor
+                } 50%)`
+              : '',
+          };
```
Comment (davidacevedo) [inline src/components/Steps/Steps.tsx:68]: I think doing this in javascript was not necessary. It's a bit complicated to read through this and I think it'd be a lot simpler if you let CSS handle this. We can most likely do this in CSS vs javascript.

Comment (davidacevedo) [inline src/components/Steps/Steps.tsx:68]: I think doing this in javascript was not necessary. It's a bit complicated to read through this and I think it'd be a lot simpler if you let CSS handle this. We can most likely do this in CSS vs javascript.

**Comment by davidacevedo** [comment]
Can you rebase the changes?

**Review by davidacevedo — APPROVED** [review]

**Review by aaronmars — APPROVED** [review]

**Pull request merged**